tests/e2e/Services/Functions/FunctionsCustomServerTest.php

```diff
@@ -2419,4 +2419,30 @@ public function testFunctionsDomainBinaryRequest()
 
         $this->assertEquals(204, $response['headers']['status-code']);
     }
+
+    public function testCreateFunctionWithResponseFormatHeader()
+    {
+        $response = $this->client->call(Client::METHOD_POST, '/functions', array_merge([
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $this->getProject()['$id'],
+            'x-appwrite-response-format' => '1.5.0', // add response format header
+        ], $this->getHeaders()), [
+            'functionId' => ID::unique(),
+            'name' => 'Test',
+            'runtime' => 'php-8.0',
+            'entrypoint' => 'index.php',
+            'timeout' => 15,
+        ]);
+
+        $this->assertEquals(201, $response['headers']['status-code']);
+
+        // Cleanup : Delete function
+        $response = $this->client->call(Client::METHOD_DELETE, '/functions/' . $response['body']['$id'], [
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $this->getProject()['$id'],
+            'x-appwrite-key' => $this->getProject()['apiKey'],
+        ], []);
+
+        $this->assertEquals(204, $response['headers']['status-code']);
+    }
```

Comment:
Khushboo what exactly is this test doing? It's not testing anything with regards to `templateVersion` or `templateBranch` ?

Reply:
When you do not specify any value for `templateBranch`, you get a server error because `content['templateBranch']` does not exist. When we add `??''` check in the code, this test starts working fine. So this test is just checking that the create function endpoint doesn't fail when no value for templateBranch is specified.

Reply:
aren't there existing tests already which do not pass templateBranch ?

Reply:
This has `'x-appwrite-response-format' => '1.5.0'` header